dask/highlevelgraph.py

```diff
@@ -814,6 +880,109 @@ def validate(self):
                     f"expected {repr(dependencies[k])}"
                 )
 
+    def __dask_distributed_pack__(self, client, client_keys) -> Any:
+        """Pack the high level graph for Scheduler -> Worker communication
+
+        The approach is to delegate the packaging to each layer in the high level graph
+        by calling .__dask_distributed_pack__() and .__dask_distributed_anno_pack__()
+        on each layer. If the layer doesn't implement packaging, we materialize the
+        layer and pack it.
+
+        Parameters
+        ----------
+        client : distributed.Client
+            The client calling this function.
+        client_keys : Iterable
+            List of keys requested by the client.
+
+        Returns
+        -------
+        data: list of header and payload
+            Packed high level graph serialized by dumps_msgpack
+        """
+        from distributed.protocol.core import dumps_msgpack
+
+        # Dump each layer (in topological order)
+        layers = []
+        for layer in (self.layers[name] for name in self._toposort_layers()):
+            layers.append(
+                {
+                    "__module__": layer.__module__,
+                    "__name__": type(layer).__name__,
+                    "state": layer.__dask_distributed_pack__(
+                        self.get_all_external_keys(),
+                        self.key_dependencies,
+                        client,
+                        client_keys,
+                    ),
+                    "annotations": layer.__dask_distributed_anno_pack__(),
+                }
+            )
+        return dumps_msgpack({"layers": layers})
+
+    @staticmethod
+    def __dask_distributed_unpack__(
+        packed_hlg, annotations: Mapping[str, Any]
+    ) -> Tuple[
+        MutableMapping[str, Any], MutableMapping[str, set], MutableMapping[str, Any]
+    ]:
```

Comment:
No reason to be overly generic in return types. The only exception is when you want to return an internal state and make it read-only.
```suggestion
    ) -> Tuple[Dict[str, Any], Dict[str, set], Dict[str, Any]]:
```